source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+
+.. code-block:: bash
+
+   singularity run --nv cuda_example.sif
+
+which will output the result of the *eigenvalues-test*, as was instructed in the definitions file under ``%runscript``. To run commands from within a shell in the container that allow for making changes, do
+
+.. code-block:: bash
+
+   singularity shell --nv --writable gpu_test.sif
+
+The container was exposed to the GPU at build-time, and at run-time it also has to be exposed with ``--nv``, otherwise it can not find the drivers! In case the container is still under development and needs debugging, use the ``--writable`` flag so that missing packages/libs can be added to the container at runtime. These packages have to be added in the definitions file for the final singularity build.
+
+.. tip::
+  
+   Only use ``--sandbox`` and ``--writable`` when developing the image. Once the build is settled, create the container with a definitions file and distribute it as-is for maximum stability.
+
+There is also a full HPC development image made available by Nvidia, called "HPC SDK", which is the software development kit that contains all the compilers, libraries and tools necessary to build efficient code that runs on GPUs. This image can be found `here <Https://catalog.ngc.nvidia.com/orgs/nvidia/containers/nvhpc>`_.
+
+Running python
+==============
+
+Popular python interfaces for modelling are tensorflow, keras, pytorch, and more. An example for using tensorflow in singularity is provided below, but some warnings have to be taken into account, due to the default behaviour of singularity with the host machine. 
+
+Starting on a machine in the GPU partition, we create a definitions file ``nv-tf-22.07.def`` containing:
+
+.. code-block:: bash
+
+  Bootstrap: docker
+  From: nvcr.io/nvidia/tensorflow:22.07-tf2-py3
+
+  %post
+  cd /tmp
+  git clone https://github.com/tensorflow/docs
+ 
+  %runscript
+  cd /tmp/docs/site/en/tutorials/keras
+  python
+ 
+  %help
+  This is a demo container to show how to run tensorflow in python
+
+and build the container using the usual 
+
+.. code-block:: bash
+
+   singularity build --nv --fakeroot nv-tf-22.07.sif nv-tf-22.07.def
+
+In this definitions file, the tensorflow docs and tutorials are installed as an example to show how to do it. 
+
+.. WARNING::
+   Running ``pip`` inside the container using ``singularity shell`` when it is in ``--writable`` mode will write the python libraries to the default **mounted** location. This location is the ``$HOME``-folder of ``$USER``. As such, pip packages will end up on the host machine and not in the container. To avoid this behaviour, only run ``pip`` during the building of the image in the definitions file, or change the mounting behaviour of singularity when entering the shell. For example, mount the local path of your project as working directory as the ``$HOME`` in the container. 
```

Comment:
Not sure if it exists already, but in the doc that explains Singularity on Spider (i.e., not here) we should mention the default binds and how to add additional binds with --bind